rgw/v2/utils/utils.py

```diff
@@ -163,46 +163,120 @@ def get_cluster_fsid():
     return cluster_fsid.rstrip("\n")
 
 
-def get_rgw_service_name():
-    rgw_orch_ls = exec_shell_cmd("sudo ceph orch ls rgw -f json-pretty")
-    rgw_service = json.loads(rgw_orch_ls)
-    rgw_service_name = rgw_service[0]["service_name"]
-    return rgw_service_name
-
+class CephOrch:
+    """
+    class for constructing ceph orch command
+    """
 
-class RGWService(object):
     def __init__(self):
         pass
 
+    def cmd(self, options):
+        """
+
+        Args:
+            options ([list]): list of options for the command
+
+        Returns:
+            str: fully constructed ceph orch command
+        """
+        options = " ".join(options)
+        log.info(f"forming ceph orch command, options: {options}")
+        cmd = f"sudo ceph orch {options}"
+        return cmd
+
+
+class SystemCTL:
+    """
+    class for constructing systemctl command
+    """
+
+    def __init__(self, unit="ceph-radosgw.target"):
+        self.unit = unit
+
+    def cmd(self, option):
+        """
+
+        Args:
+            option (str): supports start | stop | restart
+
+        Returns:
+            str: fully constructed systemctl command
+        """
+        return f"sudo systemctl {option} {self.unit}"
+
+
+class CephOrchRGWSrv:
+    """
+    class which constructs ceph orch rgw sevice
+    """
+
+    def __init__(self):
+        self.ceph_orch = CephOrch()
+        self.unit = self._unit
+
+    @property
+    def _unit(self):
+        """
+        get the service unit name
+        """
+        options = ["ls", "rgw", "-f", "json-pretty"]
```

Comment:
IMO, we could switch to using `json` instead of `json-pretty`. It allows us to avoid unnecessary characters and whitespace. JSON.loads takes care of the whitespace...

Reply:
you are right. i just tried that. but noticed that `json-pretty` output is more readable than `json`. `json-pretty` is actually prettier :) . it may help in debugging if we ever need. what do you think ?

Reply:
changed to `json`